### PR TITLE
Update pluginManagement documentation to mention global configuration options

### PR DESCRIPTION
**Pull request by apottere**
https://github.com/gradle/gradle/issues/4920

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] ~Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective~
- [x] ~Provide unit tests (under `<subproject>/src/test`) to verify logic~
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation
- [x] Recognize contributor in release notes

### CODE REVIEW

**Comment by apottere** [comment]
I wasn't sure how to format the code snippets filenames since they weren't included from source files, so I put their name in bold above the snippet.  Let me know if that should be changed.

**Comment by apottere** [comment]
I'm also not sure if the link I added is working correctly, I attempted to copy the format of the other local links in the file.

File: subprojects/docs/src/docs/userguide/plugins.adoc [hunk 1]
```diff
@@ -193,13 +193,41 @@ BUILD SUCCEEDED
...
+    // ...
+  }
+}
+----
+
+*~/.gradle/init.gradle*
+[source,groovy]
+----
+settingsEvaluated { settings ->
+  settings.pluginManagement.resolutionStrategy {
```
Comment (oehme) [inline subprojects/docs/src/docs/userguide/plugins.adoc]: Can you change this to just one block like the example above? That would look just a bit more idiomatic. :)

**Comment by apottere** [comment]
@oehme I converted the blocks to snippets, and I've verified they get copied to the `build/samples` directory when running `:docs:userguide`.  Does that address your comment or was there something else I need to change also?

**Comment by apottere** [comment]
@oehme ah, well in that case I updated the snippets to remove empty lines - I wanted to make them snippets to match the other files in the docs anyway.

**Comment by oehme** [comment]
Thank you @apottere!

**Comment by apottere** [comment]
@oehme NP!  I noticed this commit (https://github.com/gradle/gradle/pull/4999/commits/069f30f9caef9e77cd01a92b010ca50dcfbc2d18) you added before pulling - did you test those changes?  In 4.6 at least the block can't be written that way - the only way to call those specific blocks in an init script is to call
```
settings.pluginManagement.resolutionStrategy { }
```
as opposed to
```
settings.pluginManagement {
    resolutionStrategy {
    }
}```

**Comment by oehme** [comment]
I'll check it out, though that would be surprising, as both are calling the exact same method.

**Comment by apottere** [comment]
@oehme Sorry, I guess I was mistaken - it does work both ways.  I don't remember why I thought it didn't.

**Pull request merged**